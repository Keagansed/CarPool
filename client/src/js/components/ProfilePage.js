import { observer } from "mobx-react";
import React, { Component } from 'react';

import  "../../css/components/Spinner.css"

import DetailsTab from './profilepage/DetailsTab';
import { getFromStorage } from '../utils/localStorage.js';
import LoginStore from '../stores/LoginStore';
import Navbar from './Navbar'
import ProfileHeader from './profilepage/ProfileHeader';
import ProfileTabs from "./profilepage/ProfileTabs";
import TripsTab from './profilepage/TripsTab';
import VouchesTab from './profilepage/VouchesTab';


@observer class ProfilePage extends Component {
	constructor()
	{
		super();
		//========= Properties ===========
		this.state = {
		};
	}
	
	componentDidMount()//every load
	{

	}
	
	componentWillMount()// once
	{
		const obj = getFromStorage('sessionKey');
		if(obj && obj.token)
		{
			const { token } = obj;

            fetch('/api/account/verify?token='+token)
            .then(res => res.json())
            .then(json => {
                if(json.success){
					this.props.store.getProfile(token);
                }
            })
        }
	}

	handleLogout = () =>
	{
		LoginStore.setLoggedIn(false);
		LoginStore.setToken('');
		LoginStore.logOut();
	}

	
	setTab = () =>
	{
		const { store } = this.props;

		if(store.profileFound === true)
		{
			if(store.tripsTab === true)
			{
				return <TripsTab store={store}/>;
			}
			else if(store.vouchesTab === true)
			{
				return <VouchesTab store={store}/>;
			}
			else if(store.detailsTab === true)
			{
				return <DetailsTab store={store}/>;
			}
		}
		else
        {
            return(
                <div>
					<div className="spinner">
						<div className="double-bounce1"></div>
						<div className="double-bounce2"></div>
					</div>
				</div>
            )
        }
	}
	
	//render function
	render()
	{
		const { store } = this.props;

		return(
			<div className="wrapper">
				<div className="m-0 p-0 h-35">
					<div className="p-0 m-0 container-fluid">
						<div className="row m-0">

							<ProfileHeader store={store}/>

							<div className="col-md-12 p-0">
								
								<ProfileTabs store={store} />

								<div className="tab-content">
									<div className="tab-pane show active">
										{this.setTab()}
									</div>
								</div>
							</div>
						</div>
					</div>
				</div>
				<Navbar/>
			</div>
		);
		

	}
}

export default ProfilePage;
