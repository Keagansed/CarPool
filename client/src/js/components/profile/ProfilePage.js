import { observer } from "mobx-react";
import React, { Component } from 'react';

// import "./../../utils/fileQuery.js";
import LoginStore from './../../stores/LoginStore';
import Vouches from './vouchesPage/Vouches';
import Trusts from './trustPage/Trusts';
import VouchAverage from "./../vouching/VouchAverage";
import Navbar from './../navbar/Navbar';
import NavTabs from './NavTabs';
import { getFromStorage } from './../../utils/localStorage.js';

@observer class ProfilePage extends Component {
	constructor()
	{
		super();
		//========= Properties ===========
		this.state = {
			_id:""
		};
	}
	
	componentDidMount()//every load
	{
		this.props.store.getProfile(this.props.match.params._id);
		fetch('/api/account/verify?token='+this.props.match.params._id)
		.then(res => res.json())
		.then(json => {
			if(!json.success){
				this.disableEditBut();
			}
		});
	}
	
	componentWillMount()// once
	{
		const { match: {params}} = this.props;

		this.setState({
			_id: params._id,
		})

		this.props.store.getProfile(params._id);
		
		const obj = getFromStorage('sessionKey');
        if(obj && obj.token){
            const { token } = obj;
            fetch('/api/account/verify?token='+token)
            .then(res => res.json())
            .then(json => {
                if(json.success){
                    this.props.store.token = token;

                    this.setState({
                        loading: false,
                    })
                }
            })
        }
	}

	toggleEditMode()
	{
		const { store } = this.props;

		if(store.editMode)
		{
			store.editMode = false;
		}
		else
		{
			store.editMode = true;
			store.eFName = store.firstName;
			store.eLName = store.lastName;
			store.eEmail = store.email;
			store.eID = store.idNum;
		}
	}
	
	bEditSubmit = ()=>{
		const { store } = this.props;
		store.editSubmit();
	}
	
	disableEditBut = ()=>{
		this.setState({editButton:"hidden"});
	}
	
	//change handlers
	fNameChange = (e)=>{
		this.props.store.eFName = e.target.value;
	}
	lNameChange = (e)=>{
		this.props.store.eLName = e.target.value;
	}
	emailChange = (e)=>{
		this.props.store.eEmail = e.target.value;
	}
	idChange = (e)=>{
		this.props.store.eID = e.target.value;
	}
	passChange = (e)=>{
		this.props.store.ePass = e.target.value;
	}
	newPassChange = (e)=>{
		this.props.store.eNewPass = e.target.value;
	}

	handleLogout = () =>
	{
		LoginStore.setLoggedIn(false);
		LoginStore.setToken('');
		LoginStore.logOut();
	}

	uploadProfilePic = (event) =>
    {
		const { store } = this.props;

        const formData = new FormData();
        formData.append('id', this.props.match.params._id);
        formData.append('file', event.target.files[0]);

        const xhr = new XMLHttpRequest();

        xhr.open('POST', '/api/account/uploadFile/profilePicture', true);
        xhr.onreadystatechange = res =>
        {            
            if(xhr.readyState === XMLHttpRequest.DONE)
            {
                store.getProfile(this.state._id);
            }
        }

		store.editMode = false;
        xhr.send(formData);  
	}
	
	setTab = () => {
        const { store } = this.props;

        if(!this.state.loading)
        {
            if(store.vouchTab === true)
            {
                return <Vouches _id={this.state._id}/>;
            }
            else if(store.trustTab === true)
            {
                return <Trusts/>;
            }
        }

    }
	
	//render function
	render() {
		const { firstName, lastName, profilePic, secLvl } = this.props.store;
		const profilePicture = "./../api/account/getImage?filename=" + profilePic;

		if (this.props.store.profileFound)
		{
			const token = this.state._id;
			return(
				<div className="size-100 bg-purple">
					<div className="container-fluid fixed-top bg-purple">
						<div className="row height-150px bg-purple">
							<img src={profilePicture} id="profilePic" className="mx-auto my-auto rounded-circle bord-5px-white" height="120" width="120" alt="s" />
						</div>
						<div className="row height-40px bg-purple">
							<h4 className="mx-auto my-auto txt-white mbottom-0">{firstName} {lastName}</h4>
						</div>
						<div className="row height-60px bg-purple padbot-10px">
								<div className="col-6 bordright-1px-white my-auto">
									<div className="col-12 txt-center txt-white">
										<h6 className="mbottom-0"><VouchAverage _id={this.state._id}/> <i className="fa fa-star txt-gold txt-15px"></i></h6>
									</div>
									<div className="col-12 txt-center txt-white">
										<h6>Vouch Average</h6>
									</div>
								</div>
								<div className="col-6 bordleft-1px-white my-auto">
									<div className="col-12 txt-center txt-white">
										<h6 className="mbottom-0">{secLvl}/5</h6>
									</div>
									<div className="col-12 txt-center txt-white">
										<h6>Trust Rating</h6>
									</div>
								</div>
						</div>
						<div className="row">
							<NavTabs store={this.props.store} token={token}/>
						</div>
					</div>
                    <div className="padtop-300px padbot-50px">
						{this.setTab()}
					</div>
                    <Navbar token={token}/>
            	</div>
			);
		}
		else
		{
			return (
				<div>
					<h1>LOADING</h1>
				</div>
			);
		}
	}
}

export default ProfilePage;
