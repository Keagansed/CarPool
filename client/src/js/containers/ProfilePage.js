// File Type: Component

import { observer } from "mobx-react";
import React, { Component } from 'react';

import Navbar from '../components/navigation/Navbar';
import { ProfileNavTabs } from '../components/navigation/NavTabs';
import Trusts from '../components/profile/Trusts';
import Vouches from '../components/vouching/Vouches';
import VouchAverage from "../components/vouching/VouchAverage";

import { getFromStorage } from '../utils/localStorage.js';

/*
* Purpose: Container compenent for the user ProfilePage, only takes care of tab switching all data
* is in sub-components
*/
@observer class ProfilePage extends Component {
	constructor() {
		super();
	
		this.state = {
			token: ""
		};
	}

	componentWillMount() {
		let obj = getFromStorage('sessionKey');
		let token;
		
		if(obj) {
			token = obj.token;
		}

		this.setState({
			token,
		})

		this.props.store.getProfile(token);
	}
	
	/*
    * Purpose: Changes the component rendered based on which tab from NavTabs has been selected
    */
	setTab = () => {
        const { store } = this.props;

		if(store.vouchTab === true) {
			return <Vouches _id={this.state.token}/>;
		}
		else if(store.trustTab === true) {
			return <Trusts/>;
		}
	}
	
	renderProfile = () => {
		if (this.props.store.profileFound) {			
			const { firstName, lastName, profilePic, secLvl } = this.props.store;
			const profilePicture = "./../api/account/getImage?filename=" + profilePic;
			const token = this.state.token;

			return (
				<div>
					<div className="container-fluid fixed-top bg-purple">
						<div className="row height-150px bg-purple">
							<img src={profilePicture} id="profilePic" className="mx-auto my-auto rounded-circle bord-5px-white" height="120" width="120" alt="s" />
						</div>
						<div className="row height-40px bg-purple">
							<h4 className="mx-auto my-auto txt-white mbottom-0">{firstName} {lastName}</h4>
						</div>
						<div className="row height-60px bg-purple padbot-10px">
								<div className="col-6 bordright-1px-white my-auto">
									<div className="col-12 txt-center txt-white">
										<h6 className="mbottom-0"><VouchAverage _id={token}/> 
											<i className="fa fa-star txt-gold txt-15px"></i>
										</h6>
									</div>
									<div className="col-12 txt-center txt-white">
										<h6>
											Vouch Average
										</h6>
									</div>
								</div>
								<div className="col-6 bordleft-1px-white my-auto">
									<div className="col-12 txt-center txt-white">
										<h6 className="mbottom-0">
											{secLvl}/5
										</h6>
									</div>
									<div className="col-12 txt-center txt-white">
										<h6>
											Trust Rating
										</h6>
									</div>
								</div>
						</div>
						<div className="row">
							<ProfileNavTabs store={this.props.store} token={token}/>
						</div>
					</div>
					<div className="padtop-300px padbot-50px">
						{this.setTab()}
					</div>
				</div>		
			);
		} else {
			return (
				<div className="spinner">
                    <div className="double-bounce1"></div>
                    <div className="double-bounce2"></div>
                </div>
			);
		}
	}
	
	render() {
		const token = this.state.token;

		return(
			<div className="size-100 bg-purple">
				{this.renderProfile()}
				<Navbar token={token}/>
			</div>
		);	
	}
}

export default ProfilePage;
