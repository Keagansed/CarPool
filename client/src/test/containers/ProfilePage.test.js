import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import ProfilePage from '../../js/containers/ProfilePage';
import renderer from 'react-test-renderer';

class MockStore {
    profileFound = true;
    firstName = "John";
    lastName = "Doe";
    secLvl = 1.0;
    profilePic = "";

    getProfile = (token, userId) => {
        profileFound = true;
    }
}
// Shallow render does not return anthing, but tests pass. Snapshot is null.
describe('Profile Page Component', () => {
    let mockStore, match, container;

    beforeEach(() => {
        mockStore =  new MockStore();
        match = {
                params : {
                    _id : "testID"
                }
            };

        container = shallow(<MemoryRouter><ProfilePage store={mockStore} /></MemoryRouter>);
    });

    it('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><ProfilePage match={match} store={mockStore} /></MemoryRouter>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it("always renders a div", () => {
        expect(container.length).toBeGreaterThan(0);
    });

});