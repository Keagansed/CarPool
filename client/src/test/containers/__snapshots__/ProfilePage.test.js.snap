// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page Component captures snapshot 1`] = `null`;
